includes/Core/Authentication/Credentials.php

```diff
@@ -113,4 +113,21 @@ protected function get_default() {
 			'oauth2_client_secret' => '',
 		);
 	}
+
+	/**
+	 * Determines whether the authentication proxy is used.
+	 *
+	 * In order to streamline the setup and authentication flow, the plugin uses a proxy mechanism based on an external
+	 * service. This can be overridden by providing actual GCP credentials with the {@see 'googlesitekit_oauth_secret'}
+	 * filter.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool True if proxy authentication is used, false otherwise.
+	 */
+	public function using_proxy() {
+		$creds = $this->get();
+
+		return ! preg_match( '/\.apps\.googleusercontent\.com$/', $creds['oauth2_client_id'] );
```

Comment:
Let's update this as we discussed.

* If there is no `$creds['oauth2_client_id']`, return `true`
* Update the matching logic used here to check if the client ID is a proxy site ID (same as IB)

Reply:
Fixed